File: src/net/smrq/board_game_storage/shared.clj
(ns net.smrq.board-game-storage.shared
  (:require [scad-clj.scad :as scad]
            [scad-clj.model :refer :all :exclude [import use]]))

(defn cube-z0 [x y z]
  (translate [0 0 (/ z 2)] (cube x y z :center true)))

(defn extrude-linear-y [keys block]
  (->>
   (extrude-linear keys block)
   (rotate [(/ Math/PI 2) 0 (/ Math/PI 2)])))

(defn write-scad-file [file model]
  (spit file (scad/write-scad model)))

File: src/shared.js
const { modules: {
	cube,
	linear_extrude,
	rotate,
	translate,
}} = require('@smrq/openscad-js');

function cube_z0(x, y, z) {
	return translate([0, 0, z/2])(
		cube([x, y, z], { center: true })
	);
}

function linear_extrude_y(...keys) {
	return (...children) => (
		rotate([90, 0, 90])(
			linear_extrude(...keys)(...children)
		)
	);
}

module.exports = {
	cube_z0,
	linear_extrude_y,
};
